src/styles/getTooltipStyles.js

```diff
@@ -0,0 +1,117 @@
+import {Dimensions} from 'react-native';
+import spacing from './utilities/spacing';
+import styles from './styles';
+import themeColors from './themes/default';
+import fontFamily from './fontFamily';
+import variables from './variables';
+
+const GUTTER_WIDTH = 16;
+
+/**
+ * The Expensify.cash repo is very consistent about doing spacing in multiples of 4.
+ * In an effort to maintain that consistency, we'll make sure that any distance we're shifting the tooltip
+ * is a multiple of 4.
+ *
+ * @param {Number} n
+ * @returns {Number}
+ */
+function roundToNearestMultipleOfFour(n) {
+    if (n > 0) {
+        return Math.ceil(n / 4.0) * 4;
+    }
+
+    if (n < 0) {
+        return Math.floor(n / 4.0) * 4;
+    }
+
+    return 0;
+}
+
+export default function getTooltipStyles(
+    interpolatedSize,
+    xOffset,
+    yOffset,
+    width,
+    height,
+    tooltipWidth,
+    tooltipHeight,
+) {
+    // Determine if the tooltip should display below the wrapped component.
+    const windowWidth = Dimensions.get('window').width;
+    const shouldShowBelow = (yOffset - tooltipHeight) < GUTTER_WIDTH;
+
+    // Determine if we need to shift the tooltip horizontally to prevent it
+    // from displaying too near to the edge of the screen.
+    let horizontalShift = 0;
+    const tooltipLeftEdge = (xOffset + (width / 2)) - (tooltipWidth / 2);
+    const tooltipRightEdge = (xOffset + (width / 2)) + (tooltipWidth / 2);
+    if (tooltipLeftEdge < GUTTER_WIDTH) {
+        // Tooltip is in left gutter, shift right
+        horizontalShift = GUTTER_WIDTH - tooltipLeftEdge;
+    } else if (tooltipRightEdge > (windowWidth - GUTTER_WIDTH)) {
+        // Tooltip is in right gutter, shift left
+        horizontalShift = (windowWidth - GUTTER_WIDTH) - tooltipRightEdge;
+    }
+    horizontalShift = roundToNearestMultipleOfFour(horizontalShift);
```

Comment:
Maybe naming this  `gutterOffset` would help to visualize what this value will be used for eventually. Sorta lost on how this all works.

Reply:
idk I feel like `horizonalShift` is a better name, because it's the value by which the tooltip is shifted to the left or right.

Reply:
`additionalOffsetX`?

Reply:
Alright. I personally think having a variable named `shift` doesn't read as clearly as a value because it sounds like a verb. But maybe that's just me.